### PR TITLE
[Function] Support date function: microsecond()

### PR DESCRIPTION
**Pull request by zzzzzzzs**
# Proposed changes

Issue Number: close #https://github.com/apache/doris/issues/19667

## Problem summary

Support date function: microsecond()

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

File: docs/en/docs/sql-manual/sql-functions/date-time-functions/microsecond.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+{
+    "title": "minute",
```
Comment (Gabriel39) [inline docs/en/docs/sql-manual/sql-functions/date-time-functions/microsecond.md:3]:
```suggestion
    "title": "microsecond",
```

File: docs/zh-CN/docs/sql-manual/sql-functions/date-time-functions/microsecond.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+{
+    "title": "hour",
```
Comment (Gabriel39) [inline docs/zh-CN/docs/sql-manual/sql-functions/date-time-functions/microsecond.md:3]:
```suggestion
    "title": "microsecond",
```

**Review by Gabriel39 — CHANGES_REQUESTED** [review]
We also need to implement this function in BE

**Review by dataroaring — CHANGES_REQUESTED** [review]
please add a test on data from table. select microsecond(column_a);

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
> We also need to implement this function in BE

Implemented microsecond

**Comment by zzzzzzzs** [comment]
> please add a test on data from table. select microsecond(column_a);

Test has been added

File: be/src/vec/runtime/vdatetime_value.h [hunk 3]
```diff
@@ -235,6 +235,7 @@ class VecDateTimeValue { // Now this type is a temp solution with little changes
     VecDateTimeValue()
             : _neg(0),
               _type(TIME_DATETIME),
+              _microsecond(0),
```
Comment (Gabriel39) [inline be/src/vec/runtime/vdatetime_value.h:238]: We should not support microsecond for VecDateTimeValue. We only need this for DateV2

Comment (zzzzzzzs) [reply be/src/vec/runtime/vdatetime_value.h]: removed it

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 4]
```diff
@@ -304,7 +304,7 @@ valueExpression
 datetimeUnit
     : YEAR | MONTH
     | WEEK | DAY
-    | HOUR | MINUTE | SECOND
+    | HOUR | MINUTE | SECOND | MICROSECOND
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:307]: do not add this, if u do not want to implement dateAdd / dateSub / timestampDiff function in MICROSECOND unit

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 5]
```diff
@@ -418,7 +418,7 @@ interval
     ;
 
 unitIdentifier
-    : YEAR | MONTH | WEEK | DAY | HOUR | MINUTE | SECOND
+    : YEAR | MONTH | WEEK | DAY | HOUR | MINUTE | SECOND | MICROSECOND
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:421]: do not add this, if u do not want to add MICROSECOND interval literal

Comment (zzzzzzzs) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4]: It has been removed, but I will implement it next

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 [hunk 6]
```diff
@@ -247,6 +247,7 @@ MACRO: 'MACRO';
 MAP: 'MAP';
 MATCHED: 'MATCHED';
 MERGE: 'MERGE';
+MICROSECOND: 'MICROSECOND';
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4:250]: i think we do not need to add this keyword currently?

Comment (zzzzzzzs) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4]: Remove the MICROSCOND keyword from the DorisLexer.g4 and DorisParser.g4 files. If only microsecond in DorisParser.g4 are removed, then Nereids cannot parse the SQL will appear.

**Comment by zzzzzzzs** [comment]
run buildall

**Review by Gabriel39 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4 (hunk 6 above)
Comment (morrySnow) [reply fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisLexer.g4]: yes

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.List;
+
+/**
+ * ScalarFunction 'microsecond'. This class is generated by GenerateFunction.
+ */
+public class Microsecond extends ScalarFunction
+        implements UnaryExpression, ExplicitlyCastableSignature, PropagateNullableOnDateLikeV2Args {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+            FunctionSignature.ret(IntegerType.INSTANCE).args(DateTimeV2Type.MAX),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java:42]: it is not a issue about correctness, but if u need to update code again, please change arg from `DateTimeV2Type.MAX` to `DateTimeV2Type.SYSTEM_DEFAULT`

Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java:42]: OK,I will fix it when implementing the relevant microsecond function of Nereids.

**Comment by zzzzzzzs** [comment]
run p0

**Comment by zzzzzzzs** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**